spec/services/order_cycle_distributed_variants_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe OrderCycleDistributedVariants do
+  let(:order) { double(:order) }
+  let(:distributor) { double(:distributor) }
+  let(:order_cycle) { double(:order_cycle) }  
+  let(:subject) { OrderCycleDistributedVariants.new(order_cycle, distributor) }
+  let(:product) { double(:product) }
+
+  describe "checking if an order can change to a specified new distribution" do
+    it "returns false when a variant is not available for the specified distribution" do
+      order.should_receive(:line_item_variants) { [1] }
+      subject.should_receive(:available_variants) { [] }
+      expect(subject.distributes_order_variants?(order)).to be false
+    end
+
+    it "returns true when all variants are available for the specified distribution" do
+      order.should_receive(:line_item_variants) { [1] }
+      subject.should_receive(:available_variants) { [1] }
```

Comment:
Again, better to stick to RSpec 3 syntax rather than both 2 and 3.

Reply:
:+1: